test/metabase/models/setting_test.clj

```diff
@@ -199,6 +199,31 @@
               #"Cannot initialize setting before the db is set up"
               (setting/get :test-setting-custom-init)))))))
 
+(def ^:private base-options
+  {:setter   :none
+   :type     :csv
+   :default  "base-default"})
+
+(defsetting test-base-setting
+  "Setting to test the `:feature` property of settings. This only shows up in dev."
+  :visibility :internal
+  :default "my,default"
+  :base base-options)
+
+;; Avoid a false positive from `deftest-check-parallel` due to referencing the test-base-setting! setter.
+;; Even though we only resolve the (non-existent) setter, and don't call anything, the linter flags it.
+#_{:clj-kondo/ignore [:metabase/validate-deftest]}
+(deftest ^:parallel defsetting-with-base-test
+  (testing "A setting which specifies some base options"
+    (let [setting (setting/resolve-setting :test-base-setting)]
+      (testing "Uses non-overridden base options"
+        (is (= :csv (:type setting)))
+        (is (= :none (:setter setting)))
+        (is (nil? (resolve 'test-base-setting!)))
```

Comment:
I think this test would be clearer and less brittle if we tested a single attribute with two settings, one with a top-level override and one without. Otherwise we're coupling the test to the behaviour of the setting properties more than we need to. e.g.

```
(def ^:private base-options
  {:default "base-default"})

(defsetting test-no-default-with-base-setting
  "Setting to test the `:base` property of settings. This only shows up in dev."
  :base base-options)

(defsetting test-default-with-base-setting
  "Setting to test the `:base` property of settings. This only shows up in dev."
  :visibility :internal
  :base base-options
  :default "top-level-default")

(deftest ^:parallel defsetting-with-base-test
  (testing "A setting which specifies some base options"
    (let [no-top-level-default-setting (setting/resolve-setting :test-no-default-with-base-setting)
          top-level-default-setting              (setting/resolve-setting :test-default-with-base-setting)]
      (testing "Uses base options when no top-level options are specified"
        (is (= "base-default" (:default no-top-level-default-setting)))
        (is (= "base-default" (no-top-level-default-setting))))
      (testing "Uses top-level options when specified"
        (is (= "top-level-default" (:default top-level-default-setting)))
        (is (= "top-level-default" (no-top-level-default-setting)))))))
```

Reply:
Also we don't need to use the messy kondo ignore if we can avoid using the setter

Reply:
I tried to make the test as "real world" as possible, by closely resembling the uuid-nonce case, but without depending on the actual prod definition. I do really like your suggestion of reusing the same base between those settings, I agree it adds clarity.

I very intentionally do want to test the setter though, as the code needed to be reworked to avoid emitting a setter that just throws exceptions. This was in the parent PR but I still want some test coverage.

Reply:
I broke out the setter to a separate test, as it doesn't need to test both settings, and it's nice to reduce the scope of the kondo ignore.